tests/testsuite/package.rs

```diff
@@ -5943,3 +5943,191 @@ fn workspace_with_local_and_remote_deps() {
         )
         .run();
 }
+
+#[cargo_test]
+fn registry_not_in_publish_list() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+                license = "MIT"
+                description = "foo"
+                publish = [
+                    "test"
+                ]
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("package -Zpackage-workspace")
+        .masquerade_as_nightly_cargo(&["package-workspace"])
+        .arg("--registry")
+        .arg("alternative")
```

Comment:
For myself, I didn't notice the `arg` calls which made me suspicious.

At least put the `arg` calls immediately after `cargo`.  Preferably just put it in the `cargo` call.  Most end-to-end tests do that (unless the arg is dynamic) and so people are more likely to notice it that way.